### PR TITLE
Bump Kubernetes to v1.22.0

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind dependency-change

#### What this PR does / why we need it:
Update vendored Kubernetes to the latest rc to keep up with the recent changes.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Requires https://github.com/cri-o/cri-o/pull/5110

**Comment by saschagrunert** [comment]
The golang versions have to be updated to 1.16 for:
- ci/prow/images
- ci/prow/e2e-gcp
- ci/prow/e2e-agnostic

**Comment by saschagrunert** [comment]
@haircommander what was our plan for the 3 tests not using go 1.16? :point_down:

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,7 +9,6 @@ on:
   pull_request:
 env:
   GO_VERSION: '1.16'
-  GO_VERSION_LEGACY: '1.14'
```
Comment (haircommander) [inline .github/workflows/test.yml:12]: do we want to add 1.17 as well?

**Comment by haircommander** [comment]
> @haircommander what was our plan for the 3 tests not using go 1.16? point_down

I think it'll need https://github.com/openshift/release/pull/20939 and maybe some other tweaks.

File: .github/workflows/test.yml (hunk 1 above)
Comment (saschagrunert) [reply .github/workflows/test.yml:12]: I assume that go 1.17 will be released in the next few weeks. Therefore I think we can switch from 1.16 directly to 1.17. Do you have thoughts on that?

**Comment by saschagrunert** [comment]
@haircommander PTAL

File: .github/workflows/test.yml (hunk 1 above)
Comment (haircommander) [reply .github/workflows/test.yml:12]: yeah that sounds good to me

**Pull request merged**